sqlglot/optimizer/qualify_tables.py

```diff
@@ -152,9 +169,11 @@ def _qualify(table: exp.Table) -> None:
 
         for table in scope.tables:
             if not table.alias and isinstance(table.parent, (exp.From, exp.Join)):
-                exp.alias_(table, table.name, copy=False, table=True)
+                _set_alias(table, canonical_aliases, target_alias=table.name)
+
+        for column in local_columns:
```

Comment:
Note this change vs how we previously iterated over `scope.columns`. If we visited external columns in this loop, we would mutate their source too early, without having had the chance to alias outer scopes' sources.

Before making this change, the test case I added in `qualify_tables.sql` failed due to having columns qualified with `""`.